### ISSUE TITLE
Single auth - double sign-in

### ISSUE THREAD

**Issue by eljeffeg** [CybercentreCanada/assemblyline#287]
**Is your feature request related to a problem? Please describe.**
We only support SAML authentication.  When we hit the AssemblyLine page, we are met with this screen.

<img width="420" alt="Screenshot 2024-11-20 at 11 52 47 AM" src="https://github.com/user-attachments/assets/6fdc8846-19fc-45b3-ab51-0a947cac0311">

After you go through the SAML authentication, it takes you back to this unnecessary step, prompting you to "sign in".. again?

<img width="404" alt="Screenshot 2024-11-20 at 11 53 30 AM" src="https://github.com/user-attachments/assets/88c33565-a1f5-4737-9048-9bbcc8f1b2c8">

Once you click that, it signs you in.

**Describe the solution you'd like**
Just sign them in and forgo the second "sign-in" prompt when there is only one authentication method enabled.

**Comment by cccs-rs** [comment]
Hmm... could this be an issue if let's say you have access to multiple accounts using SAML/any OAuth service.

In a more general case, you'll likely only have one but I see the second sign-in as more of a confirmation of "Is this the user you want to sign in as or as a different user".

**Comment by eljeffeg** [comment]
Probably a low probability, but they could always just log out and sign in with the correct account in that situation.  It's just sort of odd that it confirms the account I just logged into.  Just log me in.  If I didn't want that account, I should have logged in with the correct account - that's not on AssemblyLine, that's on me.

**Comment by cccs-rs** [comment]
So the patch I've made so far will still display the first prompt with "Sign in with {provider}" but if there's only one provider configured on the system, it won't display the second prompt for confirmation before login.

You can test this with the [latest dev release](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.5.1.dev375).

**Comment by cccs-rs** [comment]
This should be patched in the [4.5.0.63](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.5.0.stable63) release.

**Comment by eljeffeg** [comment]
I'm at v 4.5.0.68, but I still get the double login.  My config looks like this with just saml enabled.  Do I need to put in entries for oath and set enabled: false?

```yaml
configuration:
  auth:
    internal:
      # Disable internal login, you could also leave it on if you want
      enabled: false
    saml:
      # Should SAML authentication be enabled or not
      enabled: true
      ...
 ```
